src/schema/firefox-schemas-import.js

```diff
@@ -353,8 +355,29 @@ export function rewriteExtend(schemas, schemaId) {
     (extendSchema.types || []).forEach((type) => {
       const { $extend, id, ...rest } = type;
       if ($extend) {
+        // Throw an explicit error if we are going to deep merge a set of properties which may be
+        // overriding an existing one, instead of extending the type with new ones as expected.
+        if (rest.properties) {
+          const newProps = Object.keys(rest.properties);
+          if (
+            !newProps.every(
+              (k) => extendDefinitions?.[$extend]?.properties?.[k] == null
+            )
+          ) {
+            throw new Error(
+              `Unsupported schema format: detected multiple extend schema entries overwriting existing properties while processing "${schemaId}" namespace`
+            );
```

Comment:
Import `oneLine` from `common-tags`, then re-format more or less like this:

```suggestion
            throw new Error(oneLine`Unsupported schema format: detected
              multiple extend schema entries overwriting existing properties
              while processing "${schemaId}" namespace`);
```